scripts: tests: twister_blackbox: Add more tests test_printouts.py

KamilxPaszkiet opened the pull request:
add tests to test_printouts.py:
  - timestamps
  - broken_parameter
  - help (dummy check)
  - force_color (dummy check)

KamilxPaszkiet committed:
scripts: tests: twister_blackbox: Add more tests test_printouts.py

add tests to test_printouts.py:
  -timestamps
  -broken_parameter
  -help (dummy check)
  -force_color (dummy check)

Signed-off-by: Paszkiet Kamil <kamilx.paszkiet@intel.com>

LukaszMrugala commented on scripts/tests/twister_blackbox/test_hardwaremap.py:
```diff
@@ -0,0 +1,318 @@
+                        os.remove(path)
+
+                    assert str(sys_exit.value) == '0'
+                    loggers = [logging.getLogger()] \
+                        + list(logging.Logger.manager.loggerDict.values()) \
+                        + [logging.getLogger(name) for \
+                                    name in logging.root.manager.loggerDict]
```
Previously, we've had [plus] [newline], now it's [newline] [plus] - probably should be normalised.

KamilxPaszkiet replied:
fixed!

LukaszMrugala commented on scripts/tests/twister_blackbox/test_hardwaremap.py:
```diff
@@ -0,0 +1,318 @@
+                        f'  runner: {runner}\n'+\
+                        '  serial: /dev/ttyUSB25\n'
+
+        load_data = open('test-map.yaml').read()
```
See L161

KamilxPaszkiet replied:
fixed

LukaszMrugala commented on scripts/tests/twister_blackbox/test_hardwaremap.py:
```diff
@@ -0,0 +1,318 @@
+                        '  serial: /dev/ttyUSB23\n'
+        expected_data2 = '[]\n'
+
+        load_data = open('test-map.yaml').read()
```
See L161

KamilxPaszkiet replied:
fixed

LukaszMrugala commented on scripts/tests/twister_blackbox/test_printouts.py:
```diff
@@ -166,3 +175,41 @@ def test_tree(self, capfd, test_path, expected):
+
+        output = err.split('\n')
+
+        err_lines=[]
```
Spaces around the equals sign

KamilxPaszkiet replied:
fixed

LukaszMrugala replied:
I've meant `err_lines = []` rather than `err_lines=[ ]`

KamilxPaszkiet replied:
oh, ok... thx 😃

LukaszMrugala reviewed (dismissed):
I don't see any explicit errors.
One safety question and one futureproofing change.
The rest is mostly style comments.

KamilxPaszkiet commented on scripts/tests/twister_blackbox/test_hardwaremap.py:
```diff
@@ -0,0 +1,318 @@
+                                    f'  id: {id_serial}\n'+\
+                                    '  platform: unknown\n'+\
+                                    f'  product: {id_pro}\n'+\
+                                    f'  runner: {runner}\n'+\
```
fixed

KamilxPaszkiet commented on scripts/tests/twister_blackbox/test_hardwaremap.py:
```diff
@@ -0,0 +1,318 @@
+                    product = id_pro, \
+                    serial_number = id_serial
+                    )
+                    ]
```
just to make it easier to read

nashif commented:
The help output test is just going to be a pain in the back, it does not really test any twister functionality and will fail whenever we added a new option or move things around or even change the help messages. I would remove it.

KamilxPaszkiet commented:
> The help output test is just going to be a pain in the back, it does not really test any twister functionality and will fail whenever we added a new option or move things around or even change the help messages. I would remove it.

I deleted this and left the dummy test

LukaszMrugala reviewed (approved):
Minor nitpicks, I'd merge it.
`test_help` and `test_force_color` are check-for-errors only, but I do not see a nice way for them to be fully tested.
